### PR TITLE
Removed ObjectManager usage

### PR DESCRIPTION
**Pull request by drpayyne**
### Description 

Removed unnecessary ObjectManager usage

### Manual testing scenarios
1. No testing scenario due to minor code change

### Contribution checklist 
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/PageCache/Observer/FlushCacheByTags.php [hunk 1]
```diff
@@ -40,11 +40,19 @@ class FlushCacheByTags implements ObserverInterface
...
      * @param \Magento\Framework\App\PageCache\Cache $cache
+     * @param \Magento\Framework\App\Cache\Tag\Resolver $tagResolver
+     * @param \Magento\PageCache\Model\Cache\Type $fullPageCache
      */
-    public function __construct(\Magento\PageCache\Model\Config $config, \Magento\Framework\App\PageCache\Cache $cache)
-    {
+    public function __construct(
+        \Magento\PageCache\Model\Config $config,
+        \Magento\Framework\App\PageCache\Cache $cache,
+        \Magento\Framework\App\Cache\Tag\Resolver $tagResolver,
```
Comment (atishgoswami) [inline app/code/Magento/PageCache/Observer/FlushCacheByTags.php:51]: For backward to maintain backward compatibility, you might want to change the constructor parameters a little bit.

```php
public function __construct(
    ...
    \Magento\Framework\App\Cache\Tag\Resolver $tagResolver = null,
    \Magento\PageCache\Model\Cache\Type $fullPageCache = null
) {
    ....
    $this->tagResolver   = $tagResolver ?: ObjectManager::getInstance()->get(\Magento\Framework\App\Cache\Tag\Resolver::class);
    $this->fullPageCache = $fullPageCache ?: ObjectManager::getInstance()->get(\Magento\PageCache\Model\Cache\Type::class);
}
```

Devdocs:
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/object-manager.html#exceptions

Example:
\Magento\Contact\Model\Mail::__construct

Comment (drpayyne) [reply app/code/Magento/PageCache/Observer/FlushCacheByTags.php:51]: Hi @atishgoswami, thanks for reviewing my PR, but I don't think we need to follow backward compatibility for the scope of `2.4-develop`: https://github.com/magento/magento2/pull/25875#discussion_r374904047

Comment (atishgoswami) [reply app/code/Magento/PageCache/Observer/FlushCacheByTags.php:51]: O my bad .... seems like standards have changed again ..... thanks for pointing that out ..... 
need to update certain PR's myself now 😞

File: app/code/Magento/PageCache/Test/Unit/Observer/FlushCacheByTagsTest.php [hunk 2]
```diff
@@ -32,15 +32,16 @@ protected function setUp()
...
+            $this->_cacheMock,
+            $this->tagResolver,
+            $this->fullPageCacheMock
         );
 
-        $this->tagResolver = $this->createMock(\Magento\Framework\App\Cache\Tag\Resolver::class);
-
         $helper->setBackwardCompatibleProperty($this->_model, 'tagResolver', $this->tagResolver);
         $reflection = new \ReflectionClass(\Magento\PageCache\Observer\FlushCacheByTags::class);
         $reflectionProperty = $reflection->getProperty('fullPageCache');
```
Comment (ihor-sviziev) [inline app/code/Magento/PageCache/Test/Unit/Observer/FlushCacheByTagsTest.php]: Looks like we don't need this anymore. Could you remove it?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/PageCache/Test/Unit/Observer/FlushCacheByTagsTest.php (hunk 2 above)
Comment (ihor-sviziev) [reply app/code/Magento/PageCache/Test/Unit/Observer/FlushCacheByTagsTest.php]: I meant lines 45-49. Somehow it wasn't selected

Comment (drpayyne) [reply app/code/Magento/PageCache/Test/Unit/Observer/FlushCacheByTagsTest.php]: Fixed this, thanks @ihor-sviziev!

**Review by ihor-sviziev — APPROVED** [review]
Functional test failures not related to changes from this PR. Approved.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6848](https://jira.corp.magento.com/browse/ENGCOM-6848) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
Hi @drpayyne @ihor-sviziev looks like this PR is duplicate of [#26684](https://github.com/magento/magento2/pull/26684)

**Comment by ihor-sviziev** [comment]
Yeah, this is duplicate, unfortunately I need to close this PR as duplicate of https://github.com/magento/magento2/pull/26684

**Pull request closed without merging**